cmd/integrationArtifactDownload.go

```
@@ -0,0 +1,112 @@
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runIntegrationArtifactDownload(&config, telemetryData, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runIntegrationArtifactDownload(config *integrationArtifactDownloadOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender) error {
+	clientOptions := piperhttp.ClientOptions{}
+	header := make(http.Header)
+	header.Add("Accept", "application/zip")
+	downloadArtifactURL := fmt.Sprintf("%s/api/v1/IntegrationDesigntimeArtifacts(Id='%s',Version='%s')/$value", config.Host, config.IntegrationFlowID, config.IntegrationFlowVersion)
+	tokenParameters := cpi.TokenParameters{TokenURL: config.OAuthTokenProviderURL, Username: config.Username, Password: config.Password, Client: httpClient}
+	token, err := cpi.CommonUtils.GetBearerToken(tokenParameters)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = fmt.Sprintf("Bearer %s", token)
+	httpClient.SetOptions(clientOptions)
+	httpMethod := "GET"
+	downloadResp, httpErr := httpClient.SendRequest(httpMethod, downloadArtifactURL, nil, header, nil)
+	if httpErr != nil {
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", httpMethod, downloadArtifactURL)
+	}
+	if downloadResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+	contentDisposition := downloadResp.Header.Get("Content-Disposition")
+	disposition, params, err := mime.ParseMediaType(contentDisposition)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read filename from http response headers, Content-Disposition "+disposition)
+	}
+	filename := params["filename"]
+
+	if downloadResp != nil && downloadResp.Body != nil {
+		defer downloadResp.Body.Close()
+	}
+
+	if downloadResp.StatusCode == 200 {
+		workspaceRelativePath := config.DownloadPath
+		os.MkdirAll(workspaceRelativePath, 755)
```

`os.MkdirAll` returns in error in case of a failure. I would like to suggest not so swallow that error.